exercises/practice/rest-api/src/RestApi.elm

```diff
@@ -0,0 +1,36 @@
+module RestApi exposing (buildDatabase, get, post)
+
+import Dict exposing (Dict)
+import Json.Decode exposing (Error)
+import Json.Encode
+
+
+type alias Name =
+    String
+
+
+type alias User =
+    { name : Name
+    , owes : Dict Name Int
+    , owedBy : Dict Name Int
+    , balance : Int
+    }
+
+
+type alias Database =
+    Dict Name User
+
+
+buildDatabase : String -> Result Error Database
```

Comment:
This is a difficult exercise, and if we can't change the instructions, then I think it would be worth adding some more help in this file (there is nothing in the instructions detailing what this function should do). I'm not exactly sure how best to do this, but maybe this function could be called databaseFromJsonString or something like that? Or we could supply this function ready made and ask the student to write the decoder, and then the type signature communicates what they need to do.